gsuite/resource_user.go

```diff
@@ -714,16 +654,6 @@ func resourceUserUpdate(d *schema.ResourceData, meta interface{}) error {
 		}
 	}
 
-	if d.HasChange("change_password_next_login") {
-		if v, ok := d.GetOk("change_password_next_login"); ok {
-			log.Printf("[DEBUG] Updating user change_password_next_login: %t", d.Get("change_password_next_login").(bool))
-			user.ChangePasswordAtNextLogin = v.(bool)
-		} else {
-			log.Printf("[DEBUG] Removing user change_password_next_login")
-			user.ChangePasswordAtNextLogin = false
-			nullFields = append(nullFields, "change_password_next_login")
-		}
-	}
```

Comment:
I understand the issue with this implementation, as when a user logs in this field would be set to false outside of terraform. However, it also provides a functionality to admins that allows them to use terraform to enforce a user to change their password on next login. I'm wondering if we can fix the bugs we've found with this and document (not a fan of red tape but not seeing many other options) the pitfalls of settings this to true.

Reply:
Yeah, that would be awesome but the problem is that user password reset is asynchronous.  That is, the admin trips the flag and the user, when they reset, trips it back.  So in this case terraform cannot truly be the authoritative source of truth as the state can and should change outside of the terraform flow.  If you do want terraform to be authoritative on passwords, you really have to track and manage the passwords entirely through terraform.  Obviously this poses its own problems.

Reply:
Agreed. I don't see a logical way to provide admins with a way to flip a bit once that would allow them to enforce the "change password" logic on update.